Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Types/SearchRequest.hs

```diff
@@ -45,3 +48,53 @@ data SearchRequest = SearchRequest
     createdAt :: UTCTime
   }
   deriving (Generic, PrettyShow, Show)
+
+locationMappingMakerForSearch :: (MonadFlow m) => SearchRequest -> m [DLocationMapping.LocationMapping]
```

Comment:
you can use the same function for all entities: `searchRequest`, `searchRequestSpecialZone`, `Ride`, `Booking`:

```
locationMappingMaker ::
  (MonadFlow m,
  HasField "fromLocation" entity DLoc.Location,
  HasField "toLocation" entity [DLoc.Location],
  HasField "id" entity (Id entity)
  ) =>
  entity ->
  DLocationMapping.LocationMappingTags ->
  m [DLocationMapping.LocationMapping]
locationMappingMaker entity tag = do
  let searchWithIndexes = zip ([1 ..] :: [Int]) entity.toLocation
  toLocationMappers <- mapM (locationMappingInstanceMaker entity tag) searchWithIndexes
  fromLocationMapping <- locationMappingInstanceMaker entity tag (0, entity.fromLocation)
  return $ fromLocationMapping : toLocationMappers

locationMappingInstanceMaker ::
  (MonadFlow m,
  HasField "id" entity (Id entity)
  ) =>
  entity ->
  DLocationMapping.LocationMappingTags ->
  (Int, DLoc.Location) ->
  m DLocationMapping.LocationMapping
locationMappingInstanceMaker entity tag location = do
  locationMappingId <- generateGUID
  let getIntEpochTime = round `fmap` Time.getPOSIXTime
  epochVersion <- liftIO getIntEpochTime
  let epochVersionLast5Digits = epochVersion `mod` 100000 :: Integer
  let locationMapping =
        DLocationMapping.LocationMapping
          { id = Id locationMappingId,
            tag,
            order = fst location,
            version = show epochVersionLast5Digits,
            tagId = entity.id.getId,
            location = snd location,
            ..
          }
  return locationMapping
```